src/torchpme/potentials/inversepowerlaw.py

```diff
@@ -17,6 +18,49 @@ def gamma(x: torch.Tensor) -> torch.Tensor:
     return torch.exp(gammaln(x))
 
 
+# Custom exponential function to have an autograd-compatible version of the exponential
+class CustomE1(torch.autograd.Function):
+    """The exponential integral E1(x)"""
+
+    @staticmethod
+    def forward(ctx, input):
+        ctx.save_for_backward(input)
+        if input.is_cuda:
+            input_cpu = input.cpu()
+            result = torch.tensor(exp1(input_cpu.numpy()), device=input.device)
+        else:
+            result = torch.tensor(exp1(input.numpy()))
+        return result
```

Comment:
Maybe this is a bit simpler?

```suggestion
    @staticmethod
    def forward(ctx, input):
        ctx.save_for_backward(input)
        if not input.is_cpu:
            input_numpy = input.cpu().numpy()
        else:
            input_numpy = input.numpy()

        return torch.tensor(exp1(input_numpy), device=input.device, dtype=input.dtype)
```